metals/src/main/scala/scala/meta/internal/metals/debug/DebugProvider.scala

```diff
@@ -373,6 +397,37 @@ class DebugProvider(
       }
     }
 
+  private def discoverTests(
+      id: BuildTargetIdentifier,
+      testClasses: b.ScalaTestSuites,
+  ): Future[Map[TestFramework, List[Discovered]]] = {
+    val symbolInfosList =
+      for {
+        selection <- testClasses.getSuites().asScala.toList
+        (sym, info) <- buildTargetClasses.getTestClasses(
+          selection.getClassName(),
+          id,
+        )
+      } yield compilers.info(id, sym).map(_.map(pcInfo => (info, pcInfo)))
```

Comment:
Why not use semanticdb first and only fall back to the compiler? Anyway to code needs to compile. We could keep that in TestSymbolInfo potentially if it wouldn't cause any efficiency issues.

You can use DefinitionProvider to get the file from symbol and then get semanticdb for that file

Reply:
I need to find all parents recursively. That means that for each parent symbol I'd have to recursively find the path to get the proper semanticdb. On the other hand getting info from pc should relatively cheap, since no compilation is performed here.